Heliopsis/eZFormsBundle/Tests/Provider/Form/ChainTest.php

```diff
@@ -0,0 +1,172 @@
+            ->with( $this->locations[2] )
+            ->will( $this->returnValue( $this->mockForms[2] ) );
+
+        $chainProvider = new Chain();
+
+        $chainProvider->addProvider( $this->mockProviders[0] );
+        $chainProvider->addProvider( $this->mockProviders[1] );
+        $chainProvider->addProvider( $this->mockProviders[2] );
+
+        $this->assertSame(
+            $this->mockForms[0],
+            $chainProvider->getForm( $this->locations[0] )
+        );
+
+        $this->assertSame(
+            $this->mockForms[1],
+            $chainProvider->getForm( $this->locations[1] )
+        );
+
+        $this->assertSame(
+            $this->mockForms[2],
+            $chainProvider->getForm( $this->locations[2] )
+        );
+    }
+
+    public function testGetFormChainedPriorisedProvider()
+    {
+        $this->initProviders();
+
+        $this->mockProviders[1]->expects( $this->at( 0 ) )
+            ->method( 'getForm' )
+            ->with( $this->locations[0] )
+            ->will( $this->returnValue( $this->mockForms[0] ) );
+
+        $this->mockProviders[1]->expects( $this->at( 1 ) )
+            ->method( 'getForm' )
+            ->with( $this->locations[1] )
+            ->will( $this->throwException( $this->getMock( 'Heliopsis\\eZFormsBundle\\Exceptions\\UnknownFormException' ) ) );
+
+        $this->mockProviders[1]->expects( $this->at( 2 ) )
+            ->method( 'getForm' )
+            ->with( $this->locations[2] )
+            ->will( $this->throwException( $this->getMock( 'Heliopsis\\eZFormsBundle\\Exceptions\\UnknownFormException' ) ) );
+
+        $this->mockProviders[0]->expects( $this->at( 0 ) )
+            ->method( 'getForm' )
+            ->with( $this->locations[1] )
+            ->will( $this->returnValue( $this->mockForms[1] ) );
+
+        $this->mockProviders[0]->expects( $this->at( 1 ) )
+            ->method( 'getForm' )
+            ->with( $this->locations[2] )
+            ->will( $this->throwException( $this->getMock( 'Heliopsis\\eZFormsBundle\\Exceptions\\UnknownFormException' ) ) );
+
+        $this->mockProviders[2]->expects( $this->at( 0 ) )
+            ->method( 'getForm' )
+            ->with( $this->locations[2] )
+            ->will( $this->returnValue( $this->mockForms[2] ) );
+
+        $chainProvider = new Chain();
+
+        $chainProvider->addProvider( $this->mockProviders[0], 2 );
+        $chainProvider->addProvider( $this->mockProviders[1], 1 );
+        $chainProvider->addProvider( $this->mockProviders[2], 3 );
+
+        $this->assertSame(
+            $this->mockForms[0],
+            $chainProvider->getForm( $this->locations[0] )
+        );
+
+        $this->assertSame(
+            $this->mockForms[1],
+            $chainProvider->getForm( $this->locations[1] )
+        );
+
+        $this->assertSame(
+            $this->mockForms[2],
+            $chainProvider->getForm( $this->locations[2] )
+        );
+    }
+
+    public function testNotFindFormExecption()
```

Comment:
Typo, rename to `testEmptyChainThrowsException` ?
Another test with a few non matching providers would be nice too.